app/src/main/java/com/fish/monsters/features/settings/SettingsViewModel.kt

```diff
@@ -0,0 +1,89 @@
+package com.fish.monsters.features.settings
+
+import android.app.Activity
+import android.content.Context
+import android.content.ContextWrapper
+import androidx.appcompat.app.AppCompatDelegate
+import androidx.compose.runtime.State
+import androidx.core.os.LocaleListCompat
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.fish.monsters.common.models.ui.Language
+import com.fish.monsters.common.utils.MusicManager
+import com.fish.monsters.common.utils.settings.SettingsGlobalState
+import com.fish.monsters.common.utils.settings.SettingsManager
+import com.fish.monsters.core.database.FishDatabase
+import com.fish.monsters.core.database.dao.SettingsDao
+import com.fish.monsters.core.database.entities.Settings
+import com.fish.monsters.features.settings.ui.SettingsEvent
+import kotlinx.coroutines.launch
+
+class SettingsViewModel(
+    private val settingsManager: SettingsManager,
+    private val musicManager: MusicManager,
+    private val settingsDao: SettingsDao,
+    private val fishDb: FishDatabase
+) : ViewModel() {
+    val settingsGlobalState: State<SettingsGlobalState> = settingsManager.state
+
+    fun invokeEvent(event: SettingsEvent) {
+        when (event) {
+            is SettingsEvent.ChangeLanguageEvent -> changeLanguage(event.context, event.language)
+            is SettingsEvent.SetMusicVolumeEvent -> setMusicVolume(event.percents)
+            is SettingsEvent.UpdateSettingsEvent -> {
+                with(event) {
+                    updateSettings(
+                        language, vibration, musicPercentage, soundPercentage, neonStyles
+                    )
+                }
+            }
+            is SettingsEvent.DeleteAllData -> deleteAllData()
+            SettingsEvent.SaveCurrentSettings -> saveCurrentSettings()
+        }
+    }
```

Comment:
To co tutaj widzicie może wam się wydać niepotrzebne, bo zamiast wywoływać funkcje prosto z view model przerzucamy to przez jedną funkcje której podsyłamy eventy prosto z UI po co? to jedna z dobrych praktyk w jetpack compose. Jednym benefitem z tego jest to że jest to bardzo łatwe w otestowaniu, drugim (dla nas ważniejszym) jest to że mamy dzięki temu pewność że będziemy mogli wyświelić `@Preview` . Niestety czasami jest to nie możliwe żeby podesłać sztuczny viewModel do widoku i żeby to działało, tak było w moim przypadku, Media player po prostu nie mógł być zainicjowany w `@Preview` a potrzebowałem go żeby utworzyć View model. Kolejnym benefitem jest to że wchodząc w `SettingsEvent.kt` lub patrząc na `invokeEvent()` możemy łatwo stwierdzić jakie operacje są wykonywane na danym ekranie.